lib/logstash/filters/de_dot.rb

```diff
@@ -0,0 +1,58 @@
+# encoding: utf-8
+require "logstash/filters/base"
+require "logstash/namespace"
+
+# This filter _appears_ to rename fields by replacing `.` characters with a different
+# separator.  In reality, it's a somewhat expensive filter that has to copy the
+# source field contents to a new destination field (whose name no longer contains
+# dots), and then remove the corresponding source field.
+#
+# It should only be used if no other options are available.
+class LogStash::Filters::De_dot < LogStash::Filters::Base
+
+  config_name "de_dot"
+
+  # Replace dots with this value.
+  config :separator, :validate => :string, :default => "_"
```

Comment:
Can we check if separator is not `.` again (doh) (we can throw config error). Also add a test please? :0